src/nimblepkg/nimscriptwrapper.nim

```diff
@@ -0,0 +1,131 @@
+# Copyright (C) Andreas Rumpf. All rights reserved.
+# BSD License. Look at license.txt for more info.
+
+## Implements the new configuration system for Nimble. Uses Nim as a
+## scripting language.
+
+import common, version, options, packageinfo, cli
+import hashes, json, os, streams, strutils, strtabs, tables, times, osproc, sets, pegs
+
+type
+  Flags = TableRef[string, seq[string]]
+  ExecutionResult*[T] = object
+    success*: bool
+    command*: string
+    arguments*: seq[string]
+    flags*: Flags
+    retVal*: T
+
+const
+  internalCmd = "e"
+  nimscriptApi = staticRead("nimscriptapi.nim")
+  nimscriptHash = $nimscriptApi.hash().abs()
+
+proc execNimscript(nimsFile, actionName: string, options: Options, live = true): tuple[output: string, exitCode: int] =
+  var
+    cmd = ("nim e --hints:off --verbosity:0 " & nimsFile.quoteShell & " " & actionName).strip()
+
+  if live:
```

Comment:
So the live output lets the nimscript child process write to stdout directly so you get colored output and what not. But given there's no way to get the json content from stdout simultaneously (cannot read with poParentStreams and nimscript is limited), I had to resort to the .out file.

However, the printPkgInfo task writes to stdout and nimble then writes it to the ini file so no live output for that call. I can use actionName == "printPkgInfo" instead but wanted to keep execNimscript generic. Either is fine.

Am still wondering about using json for the ini file but will comment on that separately.

Reply:
> I can use actionName == "printPkgInfo" instead but wanted to keep execNimscript generic. 

A way to make this even more generic would be to not special case this and to instead put the .ini file inside of ``retVal`` in the JSON. This would also allow you to return a typed ``ExecutionResult[string]`` from this procedure instead of the current ``string`` which can contain two different formats based on the value of `live`.

> But given there's no way to get the json content from stdout simultaneously (cannot read with poParentStreams and nimscript is limited)

I thought the plan was to output JSON amongst ordinary output and eventually deprecate the ability for Nimscript to echo, opting instead to use some sort of Nimble API. We can work on this aspect alter though, I don't mind :)

Reply:
I don't see how the returne value matters - in one case, the output is collected from a file since stdout isn't accessible, in another case, it is piped.

As for JSON to stdout, didn't realize you didn't want to show what was happening. I initially was piping everything but you could have long running processes that are building in the background and the user won't get any feedback. We can figure that out separately though.